src/walletConnect/saga.ts

```diff
@@ -331,12 +343,46 @@ function getSupportedChains() {
   })
 }
 
+export function normalizeTransaction(rawTx: any, network: Network): TransactionRequest {
+  const tx: TransactionRequest = { ...rawTx }
+
+  // Handle `gasLimit` as a misnomer for `gas`, it usually comes through in hex format
+  if ('gasLimit' in tx && tx.gas === undefined) {
+    tx.gas = isHex(tx.gasLimit) ? hexToBigInt(tx.gasLimit as Hex) : (tx.gasLimit as bigint)
+    delete tx.gasLimit
+  }
+
+  // re-calculate the feeCurrency and gas since it is not possible to tell from
+  // the request payload if the feeCurrency is set to undefined (native
+  // currency) explicitly or due to lack of feeCurrency support
+  if (network === Network.Celo) {
+    delete tx.gas
+    if ('feeCurrency' in tx) {
+      delete tx.feeCurrency
+    }
+  } else {
+    if (isHex(tx.gas)) {
+      tx.gas = hexToBigInt(tx.gas)
+    }
+  }
+
+  // Force upgrade legacy tx to EIP-1559/CIP-42/CIP-64
+  if (tx.gasPrice !== undefined) {
+    delete tx.gasPrice
+  }
+
+  // TODO: check if we need to add the nonce
```

Comment:
Yes, viem's `sendTransaction` adds it if not there, but if we're only signing, we should ensure it's set.

Reply:
thanks for clarifying, i added the nonce logic from the old normalize function back